module: fix stat with long paths on windows

targos opened the pull request:
Fix from #2011 with is an updated regression test that fails without the patch.

### v2.2.1
```
Error: Cannot find module 'D:\Documents\GitHub\io.js\test\tmp\xxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\index.js'
```

### master
```
Error: Cannot find module './other'
```

### master with fix
```

```
No error !

yosuke-furukawa commented:
CI: https://jenkins-iojs.nodesource.com/job/iojs+any-pr+multi/50/

yosuke-furukawa commented on test/parallel/test-require-long-path.js:
```diff
@@ -1,17 +1,23 @@
-var fileNameLen = Math.max(261 - common.tmpDir.length - 1, 1);
-var fileName = path.join(common.tmpDir, new Array(fileNameLen + 1).join('x'));
-var fullPath = path.resolve(fileName);
+const dirNameLen = Math.max(261 - common.tmpDir.length - 1, 1);
```
IMO, `260 - common.tmpDir.length` is simpler.

targos replied:
True. I changed it

yosuke-furukawa commented on test/parallel/test-require-long-path.js:
```diff
@@ -1,17 +1,23 @@
+fs.writeFileSync(indexFile, 'module.exports = require("./other");');
+fs.writeFileSync(otherFile, 'module.exports = 42;');
+
+assert.equal(require(indexFile), 42);
```
We would be better to add previous assertion to prevent regression.

`assert.equal(require(otherFile), 42);`

targos replied:
Done

targos commented:
addressed @yosuke-furukawa comments.
I removed the assert part so it is clearer that we are checking is the require call itself, not what it returns.

yosuke-furukawa commented:
LGTM

bnoordhuis commented:
LGTM but what's the reason for modifying an existing test instead of adding a new one?  Two regression tests are better than one.

targos commented:
Sorry, I thought it would be redundant.
`require(indexFile)` is like the existing test and the nested `require('./other')` is testing the new fix (it needs a parent directory with long path).
So the test for the new fix needs the other one to pass.

rvagg commented:
landed @ a4f4909
